crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -105,19 +115,87 @@ impl TransactionBlockEffects {
+        let mut status = self.native().status().clone();
+        let resolver: &PackageResolver = ctx.data_unchecked();
+
+        if let NativeExecutionStatus::Failure {
+            error:
+                ExecutionFailureStatus::MoveAbort(MoveLocation { module, .. }, _)
+                | ExecutionFailureStatus::MovePrimitiveRuntimeError(MoveLocationOpt(Some(MoveLocation {
+                    module,
+                    ..
+                }))),
+            command: Some(command_idx),
+        } = &mut status
+        {
+            // Get the Move call that this error is associated with.
+            if let Some(Command::MoveCall(ptb_call)) = self
+                .programmable_transaction()?
+                .and_then(|ptb| ptb.commands.into_iter().nth(*command_idx))
+            {
+                let module_new = module.clone();
+                // Resolve the runtime module ID in the Move abort to the storage ID of the package
+                // that the abort occured in. This is important to make sure that we look at the
+                // correct version of the module when resolving the error.
+                *module = resolver
+                    .resolve_module_id(module_new, ptb_call.package.into())
+                    .await
+                    .map_err(|e| Error::Internal(format!("Error resolving Move location: {e}")))?;
+            }
+        }
+
+        match status {
+            NativeExecutionStatus::Success => Ok(None),
 
             NativeExecutionStatus::Failure {
                 error,
                 command: None,
-            } => Some(error.to_string()),
+            } => Ok(Some(error.to_string())),
 
             NativeExecutionStatus::Failure {
                 error,
                 command: Some(command),
             } => {
+                let error = 'error: {
+                    let ExecutionFailureStatus::MoveAbort(loc, code) = &error else {
+                        break 'error error.to_string();
+                    };
+                    let fname_string = if let Some(fname) = &loc.function_name {
+                        format!(" in function '{}' ", fname)
+                    } else {
+                        " ".to_string()
+                    };
+                    match resolver
+                        .resolve_clever_error(loc.module.clone(), *code)
+                        .await
+                    {
+                        Some(CleverError::CompleteError {
+                            module_id,
+                            error_constant,
+                            source_line_number,
+                            error_identifier,
+                        }) => {
+                            let const_str = error_constant.unwrap_or_else(FBase64::encode);
+                            format!(
+                                "'{error_identifier}' from module '{}'{fname_string}at source line {source_line_number}\n{const_str}",
```

Comment:
I'm regretting the multi-line format and verbosity. In the long-term (not for this PR) we can expose each part of this message in its own structured field (Added that to our backlog https://mysten.atlassian.net/browse/RPC-141), but for now, thoughts on the following to get the verbosity down?

```
Error in 1st command, from 0x123abc...def::m::foo (line 42), 'kBigMistake': Useful context in value
```

As opposed to the current form:

```
Error in 1st command
'kBigMistake' from module '0x123abc...def::m' in function 'foo' at source line 42
Useful context in value
```

Changes being:

- No more newlines
- Coalesce package/module/function information, with smooth degradation if we don't have certain values.
- Drop the "source" from "source line", with the hope that it's implied by using the term "line" that it's a source line (I don't particularly mind on this one, it could be "source line 42", or it could be the even more terse "L42".

Reply:
love it! Will implement these changes :)